pageserver/src/pgdatadir_mapping.rs

```diff
@@ -1039,6 +1051,8 @@ impl<'a> DatadirModification<'a> {
         if !dir.xids.insert(xid) {
             anyhow::bail!("twophase file for xid {} already exists", xid);
         }
+        self.pending_directory_entries
+            .push((DirectoryKind::TwoPhase, dir.xids.len() as u64));
```

Comment:
Why did you end up using a u64? Because it's similar to file length?

Reply:
our other metrics are also u64, so I stayed with the type to not have yet another conversion. of course given the `O(n^2)` complexity, the pageserver would explode already at `u32::MAX`, but whatever.

Reply:
Yeah, sure. I was thinking of `usize` though for less `as u64` :)

Reply:
changed it to use an usize.